src/pj_init.c

```diff
@@ -452,55 +454,63 @@ pj_init_ctx(projCtx ctx, int argc, char **argv) {
         pj_ctx_set_errno (ctx, PJD_ERR_NO_ARGS);
         return 0;
     }
-    
+
+    /* count occurrences of pipelines and inits */
+    for (i = 0; i < argc; ++i) {
+        if (!strcmp (argv[i], "+proj=pipeline") || !strcmp(argv[i], "proj=pipeline") )
+                n_pipelines++;
+        if (!strncmp (argv[i], "+init=", 6) || !strncmp(argv[i], "init=", 5))
+            n_inits++;
+    }
+
+    /* can't have nested pipeline directly */
+    if (n_pipelines > 1) {
+        pj_ctx_set_errno (ctx, PJD_ERR_MALFORMED_PIPELINE);
+        return 0;
+    }
+
+    /* don't allow more than one +init in non-pipeline operations */
+    if (n_pipelines == 0 && n_inits > 1) {
+        pj_ctx_set_errno (ctx, PJD_ERR_TOO_MANY_INITS);
+        return 0;
+    }
+
```

Comment:
Is it necessary to disallow more than 1 init in classic operations? Wouldn't it be sufficient to disallow more than one in each step of a pipeline?

Reply:
Well... each step boils down to a classic operation, so enforcing a one-init-rule solves the pipelines steps without further logic. It is also what has been done previously, before pipelines were a thing. Lastly, I can't think of a good reason to have several inits in a single operation.

Reply:
But otherwise - If this works, I'll buy you beer: This is pipelines as I had intended them, but couldn't implement in an evidently safe way a year ago, and had to disallow nesting instead. This PR rocks!, @kbevers rocks!, pipelines rock!

And what about the `+step +init=foo +inv` case? works too? Then it **really** rocks!

Reply:
... and rolls :-)

Reply:
> And what about the +step +init=foo +inv case? works too? 

At least this abomination of a proj-string works without errors:

```
$ cat flaf
<boing> +proj=pipeline +step +proj=merc +step +proj=merc +inv +init=./flaf:puf

<puf> +proj=pipeline +step +proj=aea +step +proj=aea +inv

$ ./bin/cct +proj=pipeline +step +init=epsg:25833 +step +inv +init=epsg:25832 +step +inv +init=epsg:43
26 +step +init=./flaf:boing +step +init=epsg:25832
12 56 0 0
312928.560892657085788  6210141.326871711760759     0.000000000000000     0.000000000000000
```

It has both inverted inits and two sets of nested pipelines. I couldn't come up with something that didn't do what I expected, so maybe it's okay. I'll test it on a real world scenario tomorrow.

Reply:
Aaargh - proj -VC fails the pipeline test with exit 5001...

Reply:
> Friday bar has Belgian beers tomorrow ;-)
  
Will practice my Flemish immediately!

Reply:
> At least this abomination of a proj-string works without errors

Awesome!